eos-payg-csv/eos-payg-csv

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2019 Endless Solutions, Inc.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# Alternatively, the contents of this file may be used under the terms of the
+# GNU Lesser General Public License Version 2.1 or later (the "LGPL"), in
+# which case the provisions of the LGPL are applicable instead of those above.
+# If you wish to allow use of your version of this file only under the terms
+# of the LGPL, and not to allow others to use your version of this file under
+# the terms of the MPL, indicate your decision by deleting the provisions
+# above and replace them with the notice and other provisions required by the
+# LGPL. If you do not delete the provisions above, a recipient may use your
+# version of this file under the terms of either the MPL or the LGPL.
+
+import argparse
+import csv
+import os
+import shutil
+import subprocess
+import sys
+
+PERIODS = ['1d', '2d', '3d', '4d', '5d', '30d', '60d', '90d', '365d', 'infinite']
+
+def period_to_display_str(period):
+    if period == "infinite":
+        return "infinite"
+
+    days, _ = period.split("d")
+
+    if days == "1":
+        return "1 day"
+
+    return "{0} days".format(days)
+
+def key_to_filename(key):
+    return "{}.key".format(key)
+
+def rm_and_mkdir(dirname):
+    if os.path.isdir(dirname) and not os.path.islink(dirname):
+        if len(os.listdir(dirname)) == 0:
+            os.rmdir(dirname)
+        else:
+            shutil.rmtree(dirname)
+    os.mkdir(dirname)
+
+class CsvParseError(Exception):
+    pass
+
+def get_csv_rows(id_to_key_file):
+    rows = None
+    try:
+        with open(id_to_key_file, 'r') as f:
+            rows = list(csv.reader(f, quotechar='"'))
+    except OSError as ose:
+        raise CsvParseError("failed to read ACCT_KEY_CSV_FILE: {}".format(ose))
+
+    return rows
+
+def acct_csv_get_id_to_key(id_to_key_file):
+    first_row = True
+    id_to_key = {}
+
+    rows = get_csv_rows(id_to_key_file)
+    for row in rows:
+        if first_row:
+            # normalize in case there are \r characters, etc.
+            header_norm = ",".join(row).strip()
+            header_expected = 'device_id,code1,code2,code3,key'
+            if header_norm != header_expected:
+                raise CsvParseError('''Provided CSV invalid:
+expected header:
+    {}
+got header:
+    {}'''.format(header_expected, row))
+
+            first_row = False
+        else:
+            expected_col_len = 5
```

Comment:
It makes more sense to declare this and `expected_key_min` outside the for loop I think

Reply:
Fixed locally